File: translated_code/store.go
// store.go - 任务存储接口（Go 语言版本）
package services

import (
	"sync"
	
	"code-analysis/models"
)

// TaskStore 任务存储接口
type TaskStore interface {
	SaveTask(task *models.TaskResponse) error
	GetTask(taskID string) (*models.TaskResponse, error)
	ListTasks(dimension *models.Dimension, status *string, limit int) ([]*models.TaskResponse, error)
	SaveEvaluation(evaluation *models.EvaluationResult) error
	GetEvaluation(taskID string) (*models.EvaluationResult, error)
}

// InMemoryTaskStore 内存存储实现
type InMemoryTaskStore struct {
	tasks       map[string]*models.TaskResponse
	evaluations map[string]*models.EvaluationResult
	mu          sync.RWMutex
}

// NewInMemoryTaskStore 创建内存存储实例
func NewInMemoryTaskStore() *InMemoryTaskStore {
	return &InMemoryTaskStore{
		tasks:       make(map[string]*models.TaskResponse),
		evaluations: make(map[string]*models.EvaluationResult),
	}
}

// SaveTask 保存任务
func (s *InMemoryTaskStore) SaveTask(task *models.TaskResponse) error {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.tasks[task.TaskID] = task
	return nil
}

// GetTask 获取任务
func (s *InMemoryTaskStore) GetTask(taskID string) (*models.TaskResponse, error) {
	s.mu.RLock()
	defer s.mu.RUnlock()
	task, ok := s.tasks[taskID]
	if !ok {
		return nil, fmt.Errorf("task not found: %s", taskID)
	}
	return task, nil
}

// ListTasks 列出任务
func (s *InMemoryTaskStore) ListTasks(dimension *models.Dimension, status *string, limit int) ([]*models.TaskResponse, error) {
	s.mu.RLock()
	defer s.mu.RUnlock()

	var results []*models.TaskResponse
	for _, task := range s.tasks {
		// 过滤维度
		if dimension != nil && task.Dimension != *dimension {
			continue
		}
		// 过滤状态
		if status != nil && string(task.Status) != *status {
			continue
		}
		results = append(results, task)
	}

	// 限制返回数量
	if len(results) > limit {
		results = results[len(results)-limit:]
	}

	return results, nil
}

// SaveEvaluation 保存评估结果
func (s *InMemoryTaskStore) SaveEvaluation(evaluation *models.EvaluationResult) error {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.evaluations[evaluation.TaskID] = evaluation
	return nil
}

// GetEvaluation 获取评估结果
func (s *InMemoryTaskStore) GetEvaluation(taskID string) (*models.EvaluationResult, error) {
	s.mu.RLock()
	defer s.mu.RUnlock()
	eval, ok := s.evaluations[taskID]
	if !ok {
		return nil, fmt.Errorf("evaluation not found: %s", taskID)
	}
	return eval, nil
}

File: refactored_code/store.py
"""任务存储抽象层 - 重构版"""
from __future__ import annotations

import asyncio
from abc import ABC, abstractmethod
from typing import Protocol

from ..models import TaskResponse, EvaluationResult, Dimension


class TaskStore(ABC):
    """任务存储抽象接口"""

    @abstractmethod
    async def save_task(self, task: TaskResponse) -> None:
        """保存任务"""
        ...

    @abstractmethod
    async def get_task(self, task_id: str) -> TaskResponse | None:
        """获取任务"""
        ...

    @abstractmethod
    async def list_tasks(
        self, 
        dimension: Dimension | None = None, 
        status: str | None = None, 
        limit: int = 50
    ) -> list[TaskResponse]:
        """列出任务"""
        ...

    @abstractmethod
    async def save_evaluation(self, evaluation: EvaluationResult) -> None:
        """保存评估结果"""
        ...

    @abstractmethod
    async def get_evaluation(self, task_id: str) -> EvaluationResult | None:
        """获取评估结果"""
        ...


class InMemoryTaskStore(TaskStore):
    """内存存储实现（线程安全）"""

    def __init__(self):
        self._tasks: dict[str, TaskResponse] = {}
        self._evaluations: dict[str, EvaluationResult] = {}
        self._lock = asyncio.Lock()

    async def save_task(self, task: TaskResponse) -> None:
        async with self._lock:
            self._tasks[task.task_id] = task

    async def get_task(self, task_id: str) -> TaskResponse | None:
        async with self._lock:
            return self._tasks.get(task_id)

    async def list_tasks(
        self, 
        dimension: Dimension | None = None, 
        status: str | None = None, 
        limit: int = 50
    ) -> list[TaskResponse]:
        async with self._lock:
            results = list(self._tasks.values())
            if dimension:
                results = [t for t in results if t.dimension == dimension]
            if status:
                results = [t for t in results if t.status.value == status]
            return results[-limit:]

    async def save_evaluation(self, evaluation: EvaluationResult) -> None:
        async with self._lock:
            self._evaluations[evaluation.task_id] = evaluation

    async def get_evaluation(self, task_id: str) -> EvaluationResult | None:
        async with self._lock:
            return self._evaluations.get(task_id)


# 全局存储实例
_store: TaskStore | None = None


def get_store() -> TaskStore:
    """获取存储实例（单例）"""
    global _store
    if _store is None:
        _store = InMemoryTaskStore()
    return _store


def set_store(store: TaskStore) -> None:
    """设置存储实例（用于依赖注入）"""
    global _store
    _store = store
